crates/sui-bcs-json/src/lib.rs

```diff
@@ -0,0 +1,648 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use ::safer_ffi::prelude::*;
+use anyhow::{anyhow, bail, Error};
+use bcs::{from_bytes as bcs_from_bytes, to_bytes as bcs_to_bytes};
+use move_core_types::u256::U256;
+use safer_ffi::libc::size_t;
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value, Value::Array};
+use sui_types::{
+    base_types::{ObjectDigest, ObjectID, SequenceNumber, SuiAddress},
+    transaction::{Argument, CallArg, ProgrammableTransaction, TransactionData, TransactionKind},
+};
+
+/// Deserializes the data to given type T, and then serializes it as a JSON value
+fn to_json<T: Serialize + for<'a> Deserialize<'a>>(data: &[u8]) -> Result<Value, Error> {
+    let bcs: T = bcs_from_bytes(data)?;
+    Ok(json!(bcs))
```

Comment:
the json macro is generally used to construct json from literals, you may want to use [to_value](https://docs.rs/serde_json/latest/serde_json/fn.to_value.html) instead because using this macro also swallows the error turning it into a panic.

This comment applies to most usages of this macro throughout this file